mbed_host_tests/host_tests_conn_proxy/conn_proxy.py

```diff
@@ -174,11 +174,11 @@ def conn_process(event_queue, dut_event_queue, prn_lock, config):
             break
 
         # Send data to DUT
-        if not dut_event_queue.empty():
-            try:
-                (key, value, _) = dut_event_queue.get(timeout=1)
-            except QueueEmpty:
-                continue
+        try:
+            (key, value, _) = dut_event_queue.get(timeout=0)
```

Comment:
Yes. Just as it says. I think simpler way to explain could have been that either
block indefinitely if not timeout=None
or
block until a timeout or something to return.

Hence if not blocking and nothing to return then raise the exception.